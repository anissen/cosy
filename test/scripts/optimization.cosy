{
	var i = 4 + 3 + 2 + 1
	print i
}

{
	var i = 10
	if i > 3 + 4 {
		print 'yes'
	}
}

{
	var j = 'hello' + ' ' + 'world'
	print j
}

{
	var i = 4 + 3 + 'hello' + ' ' + 'world' + 2 + 1
	print i
}

{
	print 2 + 3
	print 2 - 3
	print 3 * 5
	print 30 / 3
	print 30 % 4
}

{
    print 4 < 5
    print 4 <= 5
    print 5 < 4
    print 5 <= 4
    print 4 > 5
    print 4 >= 5
    print 5 > 4
    print 5 >= 4
    print 4 >= 4
    print 4 <= 4
    print 4 == 5
    print 4 == 4
    print 4 != 5
    print 4 != 4
}

{
    print true and true
    print true or true
    print true or false
    print false or false
    print false and false
    print true and false
    print true and true or false and false
}
