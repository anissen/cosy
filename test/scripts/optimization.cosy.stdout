{
    var i = 10
    print i
}
{
    var i = 10
    if i > 7 {
        print 'yes'
    }
}
{
    var j = 'hello world'
    print j
}
{
    var i = '{7} hello' + ' ' + 'world {3}'
    print i
}
{
    print 5
    print -1
    print 15
    print 10
    print 2
}
{
    print true
    print true
    print false
    print false
    print false
    print false
    print true
    print true
    print true
    print true
    print false
    print true
    print true
    print false
}
{
    print true
    print true
    print true
    print false
    print false
    print false
    print true
}
